manager/allocator/network.go

```diff
@@ -927,8 +1019,83 @@ func (a *Allocator) allocateNode(ctx context.Context, node *api.Node, existingAd
 
 		allocated = true
 	}
+
+	// if we're only initializing existing addresses, we should stop here and
+	// not deallocate anything
+	if existingAddressesOnly {
+		return allocated
+	}
+
+	// now that we've allocated everything new, we have to remove things that
+	// do not belong. we have to do this last because we can easily roll back
+	// attachments we've allocated if something goes wrong by freeing them, but
+	// we can't roll back deallocating attachments by reacquiring them.
+
+	// we're using a trick to filter without allocating see the official go
+	// wiki on github:
+	// https://github.com/golang/go/wiki/SliceTricks#filtering-without-allocating
+	attachments := node.Attachments[:0]
+	for _, attach := range node.Attachments {
+		// for every attachment, go through every network. if the attachment
+		// belongs to one of the networks, then go to the next attachment. if
+		// no network matches
```

Comment:
Is the rest of this comment missing, or are we supposed to continue reading a few lines down?